email_test.go

```diff
@@ -62,7 +64,7 @@ func TestEmail_Send(t *testing.T) {
 	})
 	require.NoError(t, err)
 
-	expBody := "From: from@example.com\nTo: to@example.com\nSubject: subj\nContent-Transfer-Encoding: quoted-printable\nMIME-version: 1.0\nContent-Type: text/html; charset=\"UTF-8\"\nDate: Thu, 10 Feb 2022 23:33:58 +0000\n\nsome text\r\n"
+	expBody := "From: from@example.com\nTo: to@example.com\nSubject: subj\nMIME-version: 1.0\nDate: Thu, 10 Feb 2022 23:33:58 +0000\nContent-Transfer-Encoding: quoted-printable\nContent-Type: text/html; charset=\"UTF-8\"\n\nsome text\r\n\r\n"
```

Comment:
I'm not sure why and extra \r\n added to the test. Actually, I don't understand why the expected part of the test changed. This test doesn't involve attachment and keeping it passes was the goal of this test.

Reply:
This is because ```MIME-version``` header moves up and ```Content-Transfer-Encoding: quoted-printable``` moves down to body email boundary. 

In case with attachments correct email source looks like:

```
From: utkin.v.vitaliy@yandex.ru
To: "utkin.v.vitaliy@yandex.ru" <utkin.v.vitaliy@yandex.ru>
Subject: test with attachments
MIME-Version: 1.0
Date: Sat, 09 Apr 2022 13:28:31 +0500
Content-Type: multipart/mixed;
	boundary="----==--bound.93159.sas1-30406100349c.qloud-c.yandex.net"

------==--bound.93159.sas1-30406100349c.qloud-c.yandex.net
Content-Transfer-Encoding: 7bit
Content-Type: text/html

<div>test</div>
------==--bound.93159.sas1-30406100349c.qloud-c.yandex.net
Content-Disposition: attachment;
	filename="1.txt"
Content-Transfer-Encoding: base64
Content-Type: text/plain;
	name="1.txt"

dGVzdCBhdHRhY2htZW50Cg==
------==--bound.93159.sas1-30406100349c.qloud-c.yandex.net--
```

Without attachments looks like:

```
From: utkin.v.vitaliy@yandex.ru
To: "utkin.v.vitaliy@yandex.ru" <utkin.v.vitaliy@yandex.ru>
Subject: test
MIME-Version: 1.0
Date: Sat, 09 Apr 2022 13:27:07 +0500
Content-Transfer-Encoding: 7bit
Content-Type: text/html

<div>test</div>
```

In previous version ```Content-Transfer-Encoding``` was placed before ```MIME``` header

https://github.com/go-pkgz/email/blob/8fbf0590fceabe025a589b1e3473c20b275bd4b0/email.go#L191-L196

About extra ```\r\n``` after the body I didn't catch it either but without them mail app says ```Message parse error: Malformed MIME message. No headers found``` I can figure out why it happens if necessary.

Reply:
ok, that makes sense. 

regarding extra  \r\n after the body solving No headers found the issue - this sounds odd and we need to figure out why and how this even related to Malformed MIME message